etc/systemd/system-generators/zfs-mount-generator.in

```diff
@@ -97,10 +171,10 @@ After=zfs-import.target
 Wants=zfs-import.target
 
 [Mount]
-Where=${2}
-What=${1}
+Where=${mountpoint}
+What=${dataset}
 Type=zfs
-Options=zfsutil,auto
+Options=defaults${opts},zfsutil
```

Comment:
I replaced `/bin/mount` with a shim and logged its arguments when mounting a dataset. `defaults` appeared, first on that list. I'm just replicating that behavior. If it should be different there, we should open another issue because that seems like it might be very subtle.

Reply:
If I understand you correctly, previously, "defaults" was showing up in the call to mount, but wasn't in the unit's Options line. If you continue to leave it out of the Options, does it still show up to mount?

Reply:
No. I did something like

```sh
% mv /bin/mount /bin/mount.real
% cat > /bin/mount <<EOF
#!/bin/sh
echo "\$@" > /tmp/mount.log
/bin/mount.real "\$@"
EOF
% chmod +x /bin/mount
% zfs mount REDACTED/test
% cat /tmp/mount.log
--no-canonicalize -t zfs -o defaults,atime,relatime,dev,exec,rw,suid,nomand,zfsutil REDACTED/test /REDACTED/test
```
This exercises a call to `lib/libzfs/libzfs_mount.c:zfs_mount` coming from `cmd/zfs/zfs_main.c` (I believe). In any case, I want to mimic the behavior of calling `zfs mount` as closely as possible, hence the `defaults`.

Reply:
Okay, I see the rationale for that change. It still seems to me that defaults is unnecessary, since that is supposed to be a no-op by definition. But it shouldn’t hurt.